src/otx/engine/utils/auto_configurator.py

```diff
@@ -207,13 +208,16 @@ def get_datamodule(self) -> OTXDataModule | None:
         if self.data_root is None:
             return None
         self.config["data"]["config"]["data_root"] = self.data_root
-        data_config = self.config["data"]["config"].copy()
+        data_config = deepcopy(self.config["data"]["config"])
+        train_config = data_config.pop("train_subset")
+        val_config = data_config.pop("val_subset")
+        test_config = data_config.pop("test_subset")
         return OTXDataModule(
             task=self.config["data"]["task"],
             config=DataModuleConfig(
-                train_subset=SubsetConfig(**data_config.pop("train_subset")),
-                val_subset=SubsetConfig(**data_config.pop("val_subset")),
-                test_subset=SubsetConfig(**data_config.pop("test_subset")),
+                train_subset=SubsetConfig(sampler=SamplerConfig(**train_config.pop("sampler", {})), **train_config),
+                val_subset=SubsetConfig(sampler=SamplerConfig(**val_config.pop("sampler", {})), **val_config),
+                test_subset=SubsetConfig(sampler=SamplerConfig(**test_config.pop("sampler", {})), **test_config),
```

Comment:
In the CLI, the nested dataclass structure makes it difficult for objects to be partially instantiated, so I used a dataclass called `SamplerConfig` for this PR.
I think this could be removed when `DataModuleConfig` is refactored.